Wikipedia/SDImageCache+PromiseKit.swift

```diff
@@ -20,23 +21,25 @@ extension SDImageCache {
             }
         })
 
-        // make sure promise is rejected if the operation is cancelled
-        self.KVOController.observe(diskQueryOperation,
-                                   keyPath: "isCancelled",
-                                   options: NSKeyValueObservingOptions.New)
-        { _, _, change in
-            let value = change[NSKeyValueChangeNewKey] as! NSNumber
-            if value.boolValue {
-                reject(NSError(domain: NSURLErrorDomain, code: NSURLErrorCancelled, userInfo: nil))
+        if let diskQueryOperation = diskQueryOperation {
+            // make sure promise is rejected if the operation is cancelled
+            self.KVOController.observe(diskQueryOperation,
+                keyPath: "isCancelled",
```

Comment:
Didn't you have to update SDWebImage to call back when an operations cancelled? Why do you still need to observe isCancelled as well?